pkgs/build-support/bintools-wrapper/add-flags.sh

```diff
@@ -24,10 +24,10 @@ if [ "${NIX_BINTOOLS_WRAPPER_@infixSalt@_TARGET_TARGET:-}" ]; then
 fi
 
 for var in "${var_templates_list[@]}"; do
-    mangleVarList "$var" "${role_infixes[@]}"
+    mangleVarList "$var" ${role_infixes[@]+"${role_infixes[@]}"}
 done
 for var in "${var_templates_bool[@]}"; do
-    mangleVarBool "$var" "${role_infixes[@]}"
+    mangleVarBool "$var" ${role_infixes[@]+"${role_infixes[@]}"}
```

Comment:
Did you copy the `+` thing or did you find it was needed? No bash can tell the difference between empty and undefined arrays. Old bash needs your thing with `set -u` for empty and undefined alike, but new bash doesn't for empty and undefined alike.

Reply:
Yeah, got errors due to `role_infixes` being undefined. Happened while building `linuxHeaders`, though might have been poked due that in combination with some other element I'm missing. But yeah, this line failed otherwise.

Reply:
@dtzWill Yeah. Per the docs:

> When not performing substring expansion, using the form described below (e.g., ‘:-’), Bash tests for a parameter that is unset or null. Omitting the colon results in a test only for a parameter that is unset. Put another way, if the colon is included, the operator tests for both parameter’s existence and that its value is not null; if the colon is omitted, the operator tests only for existence.

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion

Reply:
TIL! Thanks! :+1: